### PR TITLE
Redact sensitive information from error messages

### PR DESCRIPTION
**Pull request by DZittersteyn**
This redacts api_key, login_id and token from the error objects.

### CODE REVIEW

**Comment by DZittersteyn** [comment]
Looks like you're relying on a gem that is not compatible with ruby 1.9.3, causing the Travis build to fail.

**Comment by martijnrusschen** [comment]
Hi Currencycloud folks, what's the status on this?

**Comment by janwilk** [comment]
that issue was fixed, however your PR is still failing due to Ruby 2.0 syntax

**Comment by DZittersteyn** [comment]
I added a rubocop config to enforce bracketed notation (`[:foo, :bar]`), instead of the default enforcement of percent notation (`%i(foo bar)`).

File: lib/currency_cloud/errors/error_utils.rb [hunk 1]
```diff
@@ -10,5 +10,18 @@ def platform
                        end
       "#{base}#{implementation}"
     end
+
+    REDACTED_PARAMS = [:api_key, :login_id, :token].freeze
+
+    def redacted_params(params)
```
Comment (cranieri) [inline lib/currency_cloud/errors/error_utils.rb]: Can you please add a unit test for this method?

File: lib/currency_cloud/errors/error_utils.rb [hunk 2]
```diff
@@ -10,5 +10,18 @@ def platform
...
+      redacted = params.dup
```
Comment (cranieri) [inline lib/currency_cloud/errors/error_utils.rb]: Could you please explain why you create a copy op params? That shouldn't be needed

Comment (DZittersteyn) [inline lib/currency_cloud/errors/error_utils.rb]: I'm in-place updating the `redacted` instance, wanted to ensure I don't update the actual `params`, not sure where those are used. if not duping, this should be `redact_params!` instead

**Comment by cranieri** [comment]
@DZittersteyn Thanks for the contribution and to address my comments. I merged the PR and published a new version of the SDK.

**Pull request merged**